File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> ans=new ArrayList<>();
        if(root==null) return ans;

        Queue<TreeNode> q=new LinkedList<>();
        q.offer(root);
        q.offer(null);

        List<Integer> temp=new ArrayList<>();
        while(!q.isEmpty()){
            TreeNode top=q.remove();
            if(top==null){
                ans.add(new ArrayList<>(temp));
                temp.clear();
                if(q.size()>0)
                q.offer(null);
            }else{
                temp.add(top.val);

                if(top.left!=null){
                    q.offer(top.left);
                }

                if(top.right!=null){
                    q.offer(top.right);
                }
            }
        }

        return ans;
    }
}

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    const ans=[];
    if(root==null) return ans;

    const queue=[root];
    let front=0;
    
    while(front<queue.length){
        const size=queue.length-front;
        const level=[];

        for(let i=0;i<size;i++){
            const TreeNode=queue[front++];

            level.push(TreeNode.val);
            if(TreeNode.left!=null) queue.push(TreeNode.left);
            if(TreeNode.right!=null) queue.push(TreeNode.right);
        }
        ans.push(level);  
    }
    return ans;
};